raster/src/test/scala/geotrellis/raster/geotiff/GeoTiffRasterSourceSpec.scala

```diff
@@ -85,6 +86,14 @@ class GeoTiffRasterSourceSpec extends AnyFunSpec with RasterMatchers with GivenW
     }
   }
 
+  it("should preserve baseTiff on resample/reproject") {
+    val resampled = source.resample((source.cols * 0.95).toInt , (source.rows * 0.95).toInt, NearestNeighbor)
+    val reprojected = resampled.reproject(WebMercator)
+    assert(source.tiff == reprojected.asInstanceOf[GeoTiffReprojectRasterSource].tiff)
+    val converted = reprojected.convert(UShortCellType)
+    assert(source.tiff == converted.asInstanceOf[GeoTiffReprojectRasterSource].tiff)
```

Comment:
The same here.

Reply:
according to docs i need `theSameInstanceAs` for reference comparison. will add.

Reply:
:heavy_check_mark: